src/unix/system.c

```diff
@@ -341,7 +344,32 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+	// Attempt to respect user's XDG_DATA_HOME environment variable
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
+		size_t xdg_unset_len = strlen(homedir) + strlen("/.local/share") + 1;
+
+		xdg_data_home_dir = malloc(xdg_unset_len);
+
+		if (xdg_data_home_dir == NULL) {
+			Sys_Error("%s:xdg_data_home_dir: malloc() failed.\n", __func__);
+		}
+	    check_snprintf = snprintf(xdg_data_home_dir, xdg_unset_len,
+		     "%s/%s", homedir, ".local/share");
+            if (check_snprintf < 0) {
+		    	Sys_Error("xdg_data_home_dir: XDG_DATA_HOME not set, and "
+					"storing the default path (${HOME}/.local/share) "
+					"failed.\n");
+	    }
+    }
+    check_snprintf = snprintf(homegamedir, sizeof(homegamedir),
+				 "%s/%s", xdg_data_home_dir, "quake2rtx");
+
+	free(xdg_data_home_dir);
+
+    if (check_snprintf < 0) {
+           Sys_Error("homegamedir: snprintf failed.\n");
```

Comment:
Please log the non-zero `check_snprintf` value

Reply:
Ah, earlier code seems to use all spaces. I'll quickly fix this.

Reply:
> I am not sure if you were aware of the other whitespace errors in the file, but I went ahead and fixed them too.

This is a mistake on my end. Code style reflects 8 spaces in each conditional.